app/Http/Controllers/Admin/AdminDashboardController.php

```diff
@@ -82,4 +84,59 @@ public function update(Request $request, CompanySubmission $submission)
 
         return redirect()->route('admin.submission', $submission);
     }
+
+    public function claims()
+    {
+        $submissions = QueryBuilder::for(
+            CompanyClaimSubmission::query()->with('user')->with('company')
+        )
+            ->defaultSort('-email')
+            ->allowedSorts([
+                AllowedSort::field('email'),
+                AllowedSort::field('created_at'),
+                AllowedSort::field('job_title'),
+            ])
+            ->cursorPaginate(20)
+            ->withQueryString();
+
+        $paginated = $submissions->toArray();
+
+        $data = CompanyClaimSubmissionData::collect($paginated['data']);
+
+        return Inertia::render('admin/claims',
+            [
+                'claims' => Inertia::merge($data),
+                'next_cursor' => $paginated['next_cursor'],
+            ]
+        );
+    }
+
+    public function viewClaim(Request $request, CompanyClaimSubmission $claim)
+    {
+        $claim->load('user', 'company');
+
+        $data = CompanyClaimSubmissionData::from($claim);
+
+        return Inertia::render('admin/claim', [
+            'claim' => $data,
+        ]);
+    }
+
+    public function updateClaim(Request $request, CompanyClaimSubmission $claim)
+    {
+        $action = $request->post('action');
+
+        if ($action === 'approve') {
+            $claim->update(['status' => 'approved']);
+            $claim->company()->update([
+                'claimed' => true,
+                'claimed_by' => $claim->user_id,
```

Comment:
I thought we were using roles to signify who claimed it, so that we can assign many people to the same company.

Also, talking about roles, you didn't assign a role to the Company User.